AppModuleHandler: Handle apps where a dot is in the middle of an executable name. re #5323

josephsl opened the pull request:
To avoid module lookup issue, a 'dot' (.) is replaced by an underscore (_).

Fixes #5323.

josephsl committed:
AppModuleHandler: Handle apps where a dot is in the middle of an executable name. re #5323

To avoid module lookup issue, a 'dot' (.) is replaced by an underscore (_).

jcsteh commented on source/appModuleHandler.py:
```diff
@@ -109,6 +109,9 @@ def getAppModuleFromProcessID(processID):
 		mod=runningTable.get(processID)
 		if not mod:
 			appName=getAppNameFromProcessID(processID)
+			# #5323: Certain executables contain dots as part of its file name.
```
nit: "part of its file name" should be "part of their file names"

jcsteh commented on source/appModuleHandler.py:
```diff
@@ -109,6 +109,9 @@ def getAppModuleFromProcessID(processID):
 		if not mod:
 			appName=getAppNameFromProcessID(processID)
+			# #5323: Certain executables contain dots as part of its file name.
+			if "." in appName:
```
I'd remove this if check. The reason is that replace has to search the string anyway and it will be a no-op if nothing is replaced, so there's no point in searching it twice.

josephsl committed:
AppModuleHandler: cleanup.

Comment from Jamie Teh (NV Access): better to do str.replace right away then doing a conditional (makes sense), also moved the comment to the appropriate place.
